fe/fe-core/src/main/java/com/starrocks/mv/MVMetaVersionRepairer.java

```diff
@@ -100,13 +105,53 @@ private static void repairBaseTableTableVersionChange(MaterializedView mv,
         MVVersionManager.updateEditLogAfterVersionMetaChanged(mv, maxChangedTableRefreshTime);
     }
 
+    /**
+     * Only repair partition infos which version and version time are matched with base table's version map.
+     * @param mv mv to repair
+     * @param table base table that has been changed(schema change or compactions)
+     * @param baseTableVersionMap base table's version map in mv's version map
+     * @param partitionRepairInfos partition infos to repair
+     * @return partition infos that need to be updated
+     */
+    private static List<MVRepairHandler.PartitionRepairInfo> filterToRepairPartitionInfos(
+            MaterializedView mv,
+            Table table,
+            Map<Long, Map<String, MaterializedView.BasePartitionInfo>> baseTableVersionMap,
+            List<MVRepairHandler.PartitionRepairInfo> partitionRepairInfos) {
+        List<MVRepairHandler.PartitionRepairInfo> needToUpdatePartitionInfos = Lists.newArrayList();
+        for (MVRepairHandler.PartitionRepairInfo info : partitionRepairInfos) {
+            if (!baseTableVersionMap.containsKey(table.getId())) {
+                LOG.info("Base table {} not found in mv {}'s version map, skip to repair", table.getName(), mv.getName());
+                continue;
+            }
+            Map<String, MaterializedView.BasePartitionInfo> partitionInfoMap = baseTableVersionMap.get(table.getId());
+            if (!partitionInfoMap.containsKey(info.getPartitionName())) {
+                LOG.info("Partition {} not found in base table {}'s version map, skip to repair", info.getPartitionName(),
+                        table.getName());
+                continue;
+            }
+            MaterializedView.BasePartitionInfo curBasePartitionInfo = partitionInfoMap.get(info.getPartitionName());
+            if (curBasePartitionInfo.getId() != info.getPartitionId()
+                    || curBasePartitionInfo.getVersion() != info.getLastVersion()) {
```

Comment:
info.getLastVersion() is always -1 ？

Reply:
`lastVersion` is always `newVersion -1` for now.
`lastVersionTime` is always -1 and not used. I remove it then.